src/arguments/CoreInteger.ts

```diff
@@ -0,0 +1,27 @@
+import type { PieceContext } from '@sapphire/pieces';
+import { UserError } from '../lib/errors/UserError';
+import { Argument, ArgumentContext } from '../lib/structures/Argument';
+import { err, ok, Result } from '../lib/utils/Result';
+import type { Awaited } from '../lib/utils/Types';
+
+export class CoreArgument extends Argument {
```

Comment:
since its named just add Integer in the name. pls?

Reply:
No, one name for all core pieces, helps identifying, and if somebody uses `Type`, they'll see `ArgumentStore<CorePiece>` instead of a million names.